google/cloud/aiplatform/models.py

```diff
@@ -1344,67 +1870,164 @@ def list(
+                value is too large, the deployment will error, but if deployment
+                succeeds then the ability to scale the model to that many replicas
+                is guaranteed (barring service outages). If traffic against the
+                deployed model increases beyond what its replicas at maximum may
+                handle, a portion of the traffic will be dropped. If this value
+                is not provided, the larger value of min_replica_count or 1 will
+                be used. If value provided is smaller than min_replica_count, it
+                will automatically be increased to be min_replica_count.
+            accelerator_type (str):
+                Optional. Hardware accelerator type. Must also set accelerator_count if used.
+                One of ACCELERATOR_TYPE_UNSPECIFIED, NVIDIA_TESLA_K80, NVIDIA_TESLA_P100,
+                NVIDIA_TESLA_V100, NVIDIA_TESLA_P4, NVIDIA_TESLA_T4
+            accelerator_count (int):
+                Optional. The number of accelerators to attach to a worker replica.
+            service_account (str):
+                The service account that the DeployedModel's container runs as. Specify the
+                email address of the service account. If this service account is not
+                specified, the container runs as a service account that doesn't have access
+                to the resource project.
+                Users deploying the Model must have the `iam.serviceAccounts.actAs`
+                permission on this service account.
+            explanation_metadata (aiplatform.explain.ExplanationMetadata):
+                Optional. Metadata describing the Model's input and output for explanation.
+                Both `explanation_metadata` and `explanation_parameters` must be
+                passed together when used. For more details, see
+                `Ref docs <http://tinyurl.com/1igh60kt>`
+            explanation_parameters (aiplatform.explain.ExplanationParameters):
+                Optional. Parameters to configure explaining for Model's predictions.
+                For more details, see `Ref docs <http://tinyurl.com/1an4zake>`
+            metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as
+                metadata.
             sync (bool):
                 Whether to execute this method synchronously. If False, this method
                 will be executed in concurrent Future and any downstream object will
                 be immediately returned and synced when the Future has completed.
         """
-        self._sync_gca_resource()
+        self._sync_gca_resource_if_skipped()
 
-        models_to_undeploy = sorted(  # Undeploy zero traffic models first
-            self._gca_resource.traffic_split.keys(),
-            key=lambda id: self._gca_resource.traffic_split[id],
+        if len(self._gca_resource.deployed_models):
```

Comment:
This usage could raise in the case where the endpoint was created with `sync=False` as the resource may not have been created. The `_sync_gca_resource_if_skipped` applies only to the special case where an already created Endpoint is initialized with it's resource name. We skip the GET call to remove the additional latency of `Endpoint(resource_name).predict(...)`.

Preference to rely on the service to throw.

Reply:
Hey Sasha, I'm a bit confused on what the fix here is, should we delete the line `self._sync_gca_resource_if_skipped()`? Isn't this line in the deploy() method, so how does it relate to predict()? Also what is the service that we have a preference towards?

Thanks,
Nishant

Reply:
We should remove this validation and allow the service to throw when the user attempts to deploy a second model to a PrivateEndpoint.

Reply:
Why was this resolved?